src/safeposix/syscalls/fs_calls.rs

```diff
@@ -961,13 +1014,8 @@ impl Cage {
             return newfd;
         }
 
-        //need to add close helper and reference
-        match fdtable.get(&newfd) {
-            Some(_) => {return Self::_close_helper(&self, newfd, Some(&fdtable), false);},
-            None => {
-                fdtable.insert(newfd, fdtable.get(&oldfd).unwrap().clone());
-            } 
-        }
+        if fdtable.contains_key(&newfd) {return Self::_close_helper(&self, newfd, Some(&fdtable), false);}    
```

Comment:
The close result shouldn't be returned, it should still add to the fdtable even if it has to close what was there previously